scripts/test-upgrade.sh

```diff
@@ -105,6 +105,41 @@ cd ${OA_DIR}/playbooks
 ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VLAN_FLAT}\" neutron"
 ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VXLAN}\" neutron"
 
+# CEPH CHANGES
+# These are the instructions to give if there is a failure message on ceph-mon.yml
+if [ $(ansible mons --list-hosts | wc -l) -gt 1 ]; then
+  #Gather facts about all the monitors first, even with bad names
+  #This shouldn't change a thing, it's a regular playbook run.
+  openstack-ansible ${RPCD_DIR}/playbooks/gen-facts.yml
+  for mon in $(ansible mons --list-hosts); do
+    ansible $mon -m command -a "stop ceph-mon id=$mon"
+    ansible $mon -m command -a "ceph mon remove $mon"
+    openstack-ansible lxc-containers-destroy.yml --skip-tags=container-directories --limit $mon
+    openstack-ansible lxc-containers-create.yml --limit $mon
+    # Destroy facts for the current mon, this way it's gonna be properly generated
+    rm -f /etc/openstack_deploy/ansible_facts/$mon
+    openstack-ansible ${RPCD_DIR}/playbooks/ceph-mon.yml --limit $mon
+  done
+
+  # During the migrate-yaml.py process, the "raw_multi_journal: true" directive
+  # was moved into the user_rpco_variables_*.yml files. This will cause the ceph playbooks
+  # to fail when validating these variables because "osd_directory: true" is also
+  # set as part of the deploy.sh process(https://github.com/rcbops/rpc-openstack/blob/23fd14af75b935ff9778f438740a45e0d6fe8679/scripts/deploy.sh#L93).
+  # "osd_directory" and "raw_multi_journal" cannot be both set to true by design:
+  # https://github.com/ceph/ceph-ansible/blob/e0173866453c363c766e8323796ef54029cb5f96/group_vars/osds.sample#L58
+  #
+  # To make sure these variables don't confict, we must make sure "raw_multi_journal" is
+  # set to false in the "user_rpco_variables_*.yml" files. NOTE: This is the intended behavior only
+  # for AIOs. It might be different for multi-node deployments.
+  sed -i "s/raw_multi_journal:.*/raw_multi_journal: false/" \
+  /etc/openstack_deploy/user_rpco_variables_defaults.yml
+  sed -i "s/raw_multi_journal:.*/raw_multi_journal: false/" \
+  /etc/openstack_deploy/user_rpco_variables_overrides.yml
```

Comment:
I see the need for this -- but I don't think we should be adding this here.  There may be many other issues resulting from using migrate-yaml.yml with --for-testing-take-new-vars-only and handling these cases in this script will make it increasingly more difficult to understand what is going on.

I believe the Jenkins job has its own overrides file, we'll need to ensure that these vars are defined there to avoid being trampled on.